app/models/setting.rb

```diff
@@ -24,6 +33,36 @@ def validate(record)
     end
   end
 
+  module IpValidationCommon
```

Comment:
Please split this and move it to app/validators, this logic isn't exactly Setting's logic